ntbs-service-unit-tests/DataMigration/CaseManagerImportServiceTests.cs

```diff
@@ -106,92 +106,6 @@ public async Task WhenCaseManagerForLegacyNotificationExistsInNtbs_UserNotImport
-            notification.TreatmentEvents =
-                new List<TreatmentEvent> {new TreatmentEvent {CaseManagerUsername = CASE_MANAGER_USERNAME_2, TbServiceCode = "TBS00TEST"}};
-
-            // Act
-            await _caseManagerImportService.ImportOrUpdateCaseManagersFromNotificationAndTreatmentEvents(notification, null, "test-request-1");
-
-            // Assert
-            var addedUsers = _context.User.ToList();
-            var addedUserFromTreatmentEvent = addedUsers.SingleOrDefault(u => u.Username == CASE_MANAGER_USERNAME_2);
-            Assert.NotEmpty(addedUsers);
-            Assert.Equal(2, addedUsers.Count);
-            Assert.NotNull(addedUserFromTreatmentEvent);
-            Assert.Equal("Martin", addedUserFromTreatmentEvent.GivenName);
-            Assert.Equal("Francis", addedUserFromTreatmentEvent.FamilyName);
-            Assert.False(addedUserFromTreatmentEvent.IsActive);
-            Assert.False(addedUserFromTreatmentEvent.IsCaseManager);
-            Assert.DoesNotContain("TBS00TEST",
-                addedUserFromTreatmentEvent.CaseManagerTbServices.Select(cmtb => cmtb.TbServiceCode));
-        }
-
-        [Fact]
-        public async Task WhenNotificationHasNoCaseManager_AddsCaseManagersFromTreatmentEvents()
-        {
-            // Arrange
-            var notification = GivenLegacyNotificationWithNoCaseManager();
-            GivenLegacyUserWithName(CASE_MANAGER_USERNAME_1, "Pietro", "Peters");
-            await GivenLegacyUserHasPermissionsForTbServiceInHospital(CASE_MANAGER_USERNAME_1, "TBS00FAKE", HOSPITAL_GUID_1);
-            GivenLegacyUserWithName(CASE_MANAGER_USERNAME_2, "Scarlett", "Violet");
-            await GivenLegacyUserHasPermissionsForTbServiceInHospital(CASE_MANAGER_USERNAME_2, "TBS00RONG", HOSPITAL_GUID_2);
-            notification.TreatmentEvents =
-                new List<TreatmentEvent>
-                {
-                    new TreatmentEvent {CaseManagerUsername = CASE_MANAGER_USERNAME_1, TbServiceCode = "TBS00FAKE"},
-                    new TreatmentEvent {CaseManagerUsername = CASE_MANAGER_USERNAME_2, TbServiceCode = "TBS00RONG"},
-                };
-
-            // Act
-            await _caseManagerImportService.ImportOrUpdateCaseManagersFromNotificationAndTreatmentEvents(notification, null, "test-request-1");
-
-            // Assert
-            var addedUsers = _context.User.ToList();
-            Assert.Equal(2, addedUsers.Count);
-            Assert.Equal("Pietro", addedUsers.Single(u => u.Username == CASE_MANAGER_USERNAME_1).GivenName);
-            Assert.Equal("Peters", addedUsers.Single(u => u.Username == CASE_MANAGER_USERNAME_1).FamilyName);
-            Assert.Equal("Scarlett", addedUsers.Single(u => u.Username == CASE_MANAGER_USERNAME_2).GivenName);
-            Assert.Equal("Violet", addedUsers.Single(u => u.Username == CASE_MANAGER_USERNAME_2).FamilyName);
-        }
-
```

Comment:
Have these tests been replaced elsewhere?

Reply:
No not really as the case manager service now always uses the same function `ImportOrUpdateLegacyUser` either if this is called directly (in NotificationMapper for treatment events) or with a notification passed in (from notification import service and here). I don't think these tests should be added into the NotificationMapper tests as it doesn't seem the right place and we'd just be testing same thing we are testing here.  I could add in tests for `ImportOrUpdateLegacyUser` but they would be basically the same tests as I've left here without creating the legacy notification

Reply:
Since the `NotificationMapper` is where the logic now resides, I think it's the only place to add these tests. What makes you hesitant?

If your concern is that it's a complicated class that's doing a lot of other things, then you could think about breaking the relevant logic into another class (e.g. `TreatmentEventMapper`) and adding tests for that.

Reply:
I have added a TreatmentEvenMapper and related tests